packages/concerto-tools/lib/codegen/fromopenapi/cto/concertometamodelvisitor.js

```diff
@@ -0,0 +1,522 @@
+
+// }
+
+function flattenOaiAnyOfOrOneOf([oaiPropertyName, oaiPropertyBody]) {
+    const alternative = oaiPropertyBody.anyOf ||
+        oaiPropertyBody.oneOf;
+
+    if (alternative) {
+        const keyword = oaiPropertyBody.anyOf ? 'anyOf' : 'oneOf';
+        console.warn(
+            `Keyword '${keyword}' in definition '${oaiPropertyName}' is not fully supported. Defaulting to first alternative.`
+        );
+
+        delete oaiPropertyBody.anyOf;
+        delete oaiPropertyBody.oneOf;
+
+        return [
+            oaiPropertyName, {
+                ...oaiPropertyBody,
+                ...alternative
+            }
+        ];
+    }
+
+    return [oaiPropertyName, oaiPropertyBody];
+}
+
+function isOaiPropertyUnmodelledJson(oaiPropertyBody) {
+    return oaiPropertyBody.type === 'object' &&
+        typeof oaiPropertyBody.additionalProperties === 'object';
+}
+
+function oaiTypeToConerto$class([oaiPropertyName, oaiPropertyBody]) {
+    if (isOaiPropertyArray(oaiPropertyBody)) {
+        return oaiTypeToConertoBasic$class([
+            oaiPropertyName,
+            oaiPropertyBody.items
+        ]);
+    }
+
+    if (isOaiPropertyObject(oaiPropertyBody)) {
+        return;
+    }
+
+    return oaiTypeToConertoBasic$class([oaiPropertyName, oaiPropertyBody]);
+}
+
+function oaiToConcertoProperty([oaiPropertyName, oaiPropertyBody]) {
+    // const oaiPropertyWithout
+    return {
+        $class: oaiTypeToConerto$class([oaiPropertyName, oaiPropertyBody]),
+        isArray: isOaiPropertyArray(oaiPropertyBody),
+        isOptional: false // TODO:
+    };
+}
+
+function oaiToConcertoProperties(oaiProperties) {
+    return Object.entries(oaiProperties)
+        .map(flattenOaiAnyOfOrOneOf)
+        // @ts-ignore
+        .map(oaiToConcertoProperty);
+}
+
+function oaiObjectToDeclaration([oaiObjectK, oaiObjectV]) {
+    const properties = oaiToConcertoProperties(oaiObjectV.properties);
+    return {
+        $class: `${parameters.metaModelNamespace}.ConceptDeclaration`,
+        name: oaiObjectK,
+        isAbstract: false,
+        properties
+    };
+}
+
+
+class Visitable {
+    accept(visitor, parameters) {
+        return visitor.visit(this, parameters);
+    }
+}
+
+class BodyVisitable extends Visitable {
+    constructor(body) {
+        super();
+        this.body = body;
+    }
+}
+
+class NameAndBodyVisitable extends Visitable {
+    constructor(name, body) {
+        super();
+        this.name = name;
+        this.body = body;
+    }
+}
+
+class OaiLocalReference extends BodyVisitable {
+    isOaiLocalReference = true;
+}
+
+class OaiReference extends BodyVisitable {
+    isOaiReference = true;
+}
+
+class OaiProperty extends NameAndBodyVisitable {
+    isOaiProperty = true;
+}
+
+class OaiProperties extends BodyVisitable {
+    isOaiProperties = true;
+}
+
+class OaiSchema extends NameAndBodyVisitable {
+    isOaiSchema = true;
+}
+
+class OaiSchemas extends BodyVisitable {
+    isOaiSchemas = true;
+}
```

Comment:
Consider moving these definitions to a new file to improve separation of concerns

Reply:
Addressed ✅